wordpress-mobile/WordPress-Android
# Issue/15639 update look and logic of reply to comment button

Fixes #15639
Fixes #15621

This PR updates the UI and logic of the Reply button in threaded comments.

[![Image from Gyazo](https://i.gyazo.com/765c3ad181e94823a2093514721031fc.gif)](https://gyazo.com/765c3ad181e94823a2093514721031fc)

To test:
- In a threaded comment,  click on a reply button.
- Confirm that the Reply button and Icon next to it are highlighted with blue color.
- Rotate the device and confirm that the highlight remains.
- Press on the Reply button again, and confirm that the highlight is gone, and the Reply Box changes from "Reply to comment" to "Reply to post"
- Press on the Reply button to select it, and press the back button. confirm that the highlight is gone, and the Reply Box changes from "Reply to comment" to "Reply to post"
- - Press on the Reply button to select it, and then reply to the comment. Confirm that the highlight is gone, and the Reply Box changes from "Reply to comment" to "Reply to post"

## Regression Notes
1. Potential unintended areas of impact
Limited to comment reply and highlight functionality.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
Legacy code is not very testable.

PR submission checklist:
